contracts/Controller.sol

```diff
@@ -454,7 +464,26 @@ contract Controller is ReentrancyGuard, Ownable {
      * @dev only account owner or operator can withdraw long option from vault
      * @param _args MintArgs structure
      */
-    // function _mintOtoken(Actions.MintArgs memory _args) internal isAuthorized(_args.owner) {}
+    function _mintOtoken(Actions.MintArgs memory _args)
+        internal
+        isAuthorized(msg.sender, _args.owner)
+        returns (MarginAccount.Vault memory)
+    {
+        require(checkVaultId(_args.owner, _args.vaultId), "Controller: invalid vault id");
+        require(_args.to == msg.sender, "Controller: minter address and msg.sender address mismatch");
+
```

Comment:
does the check on isValidOtoken happen in margin calculator? (Like we shouldn't allow someone to mint a non Opyn otoken right?)

Reply:
Yeah, I don't think calculator checks this, so need that check here. isWhitelistedOtoken(short)

Reply:
Yeah.

Reply:
Will add it.